molecule/app_deploy_domain/prepare.yml

```diff
@@ -0,0 +1,198 @@
+---
+- name: Prepare
+  hosts: all
+  gather_facts: true
+  vars_files:
+    - ../group_vars/all/vars.yml
+  vars:
+    sudo_pkg_name: 'sudo'
+  tasks:
+    - name: "Ensure {{ sudo_pkg_name }} is installed (if user is root)."
+      ansible.builtin.yum:
+        name: "{{ sudo_pkg_name }}"
+      when:
+        - ansible_user_id == 'root'
+
+    - name: Gather the package facts
+      ansible.builtin.package_facts:
+        manager: auto
+
+    - name: "Check if {{ sudo_pkg_name }} is installed."
+      ansible.builtin.assert:
+        that:
+          - sudo_pkg_name in ansible_facts.packages
+
+    - name: "Ensure required packages are installed."
+      become: "{{ wildfly_install_requires_become | default(true) }}"
+      ansible.builtin.yum:
+        name:
+          - java-17-openjdk
+          - procps-ng
+          - tzdata-java
+        state: present
+
+    - name: "Ensure files directory exists on controller"
+      ansible.builtin.file:
+        path: "{{ playbook_dir }}/../files"
+        state: directory
+        mode: '0755'
+      delegate_to: localhost
+      run_once: true
+
+    - name: "Download test application to controller"
+      ansible.builtin.get_url:
+        url: "https://github.com/rpelisse/simple-servlet/releases/download/info_wfly31_1.2/info-1.2.war"
```

Comment:
We should preferably use something from the community, for example, https://tomcat.apache.org/tomcat-11.0-doc/appdev/sample/sample.war